engine/src/test/java/io/camunda/zeebe/engine/processing/deployment/MultiPartitionDeploymentLifecycleTest.java

```diff
@@ -53,20 +54,32 @@ public void shouldTestLifecycle() {
     final var deploymentPartitionRecords =
         RecordingExporter.records().withPartitionId(1).limit(10).collect(Collectors.toList());
 
-    assertThat(deploymentPartitionRecords)
+    assertThat(deploymentPartitionRecords).hasSize(10);
```

Comment:
:thinking: Not sure this is a useful assertion. Could change with implementation details. Same with the limit call above